aea/aea_builder.py

```diff
@@ -1340,11 +1340,17 @@ def _load_and_add_components(
 
             if configuration in self._component_instances[component_type].keys():
                 component = self._component_instances[component_type][configuration]
+                if configuration.component_type != ComponentType.SKILL:
+                    component.logger = cast(
+                        logging.Logger, make_logger(configuration, agent_name)
+                    )
             else:
                 configuration = deepcopy(configuration)
-                component = load_component_from_config(configuration, **kwargs)
+                _logger = make_logger(configuration, agent_name)
+                component = load_component_from_config(
+                    configuration, logger=_logger, **kwargs
+                )
 
-            _set_logger_to_component(component, configuration, agent_name)
```

Comment:
Why was this not sufficient? Because we then don't have it in the constructor I assume?

Reply:
yes; now we pass the logger to the `Component.__init__`, so it can be used during the construction of the logger